src/libraries/System.Private.CoreLib/src/System/IO/File.cs

```diff
@@ -512,64 +511,39 @@ public static Task WriteAllTextAsync(string path, string? contents, Encoding enc
             return WriteToFileAsync(path, FileMode.Create, contents, encoding, cancellationToken);
         }
 
-        public static Task<byte[]> ReadAllBytesAsync(string path, CancellationToken cancellationToken = default(CancellationToken))
+        public static async Task<byte[]> ReadAllBytesAsync(string path, CancellationToken cancellationToken = default(CancellationToken))
         {
-            if (cancellationToken.IsCancellationRequested)
-            {
-                return Task.FromCanceled<byte[]>(cancellationToken);
-            }
+            cancellationToken.ThrowIfCancellationRequested();
```

Comment:
If cancellation has been requested, this is going to throw an exception, which will then be immediately caught.  The previous version avoided that overhead by doing the check for cancellation in a non-async method.